### PR TITLE
Feature/invoice ocr assistant

### PR DESCRIPTION
**Pull request by Stitch-z**
### Features: add a new role invoice ocr assistant. 
  - Function: Invoice OCR assistant, supporting OCR text recognition of invoice PDF, PNG, JPG, and ZIP files, generating a table of payee, city, total amount, and invoicing date for invoices (supporting multiple languages), and asking questions about individual files based on the OCR recognition results of invoices.
 - The table results generated by unit testing:
image

 - Unit tests execution results:
image
image
image
image

### CODE REVIEW

File: metagpt/actions/invoice_ocr.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        """Unzip a file and return the path to the unzipped directory.
+
+        Args:
+            file_path: The path to the zip file.
+
+        Returns:
+            The path to the unzipped directory.
+        """
+        file_directory = file_path.parent
+        with zipfile.ZipFile(file_path, 'r') as zip_ref:
```
Comment (better629) [inline metagpt/actions/invoice_ocr.py:106]: use " if you have used in other places

File: metagpt/actions/invoice_ocr.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+                # Identify files that match the type
+                if filename.split(".")[-1] in FileExtensionType.get_type_list():
+                    ocr = PaddleOCR(use_angle_cls=True, lang="ch", page_num=1)
+                    ocr_result = ocr.ocr(str(invoice_file_path), cls=True)
+                    ocr_list.append(ocr_result)
+            return ocr_list
+
+        else:
+            #  OCR identifies single file
+            ocr = PaddleOCR(use_angle_cls=True, lang="ch", page_num=1)
```
Comment (better629) [inline metagpt/actions/invoice_ocr.py:148]: suggest line 141 142 and 148 149 as an ocr function

File: metagpt/actions/invoice_ocr.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+            filename: The name of the file.
+
+        Returns:
+            The file type based on FileExtensionType enum.
+
+        Raises:
+            Exception: If the file format is not zip, pdf, png, or jpg.
+        """
+        file_ext = None
+        for ext in FileExtensionType:
```
Comment (better629) [inline metagpt/actions/invoice_ocr.py:85]: may an  support_ext = [".zip", ".jpg",xxx]  is more simple. and you can use Path(xx).suffix to have a judge. The enum make it complex.

File: metagpt/actions/invoice_ocr.py [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+        name: The name of the action. Defaults to an empty string.
+        language: The language used for the generated table. Defaults to "ch" (Chinese).
+
+    """
+
+    def __init__(self, name: str = "", language: str = "ch", *args, **kwargs):
+        super().__init__(name, *args, **kwargs)
+        self.language = language
+
+    async def run(self, ocr_results: list, filename: str, *args, **kwargs) -> Dict[str, str]:
```
Comment (better629) [inline metagpt/actions/invoice_ocr.py:166]: keep same. if use list, you can use dict

File: metagpt/roles/invoice_ocr_assistant.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+"""
+@Time    : 2023/9/21 14:10:05
+@Author  : Stitch-z
+@File    : invoice_ocr_assistant.py
+"""
+
+import os
+from pathlib import Path
+
+import pandas as pd
```
Comment (better629) [inline metagpt/roles/invoice_ocr_assistant.py:13]: add new line

File: metagpt/roles/invoice_ocr_assistant.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        profile: The role profile description.
+        goal: The goal of the role.
+        constraints: Constraints or requirements for the role.
+        language: The language in which the invoice table will be generated.
+    """
+
+    def __init__(
+        self,
+        name: str = "Stitch",
+        profile: str = "Invoice Ocr Assistant",
```
Comment (better629) [inline metagpt/roles/invoice_ocr_assistant.py:36]: OCR

File: requirements.txt [hunk 7]
```diff
@@ -1,3 +1,5 @@
+paddlepaddle==2.4.2
```
Comment (better629) [inline requirements.txt:1]: for most people, they don't need paddlexxx, maybe these requirements should in other requirexxx.txt

**Comment by better629** [comment]
LGTM

**Comment by stellaHSR** [comment]
LGTM

**Pull request merged**